aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -119,10 +119,50 @@ void index_put_kernel(TensorIterator& iter, IntArrayRef index_size, IntArrayRef
   });
 }
 
+template <typename scalar_t, typename mask_t, typename func_t>
+void cpu_masked_select_kernel(TensorIterator& iter, const func_t& f) {
+  auto is_mask_bool = std::is_same<mask_t, bool>::value;
+  auto loop = [&](char** data, const int64_t* strides, int64_t n) {
+    char* dst = data[0];
+    char* src = data[1];
+    char* mask = data[2];
+    char* mask_prefix_sum = data[3];
+    for (int64_t i = 0; i < n; i++) {
+      mask_t mask_value = *(mask_t*)(mask + strides[2] * i);
+      if (!is_mask_bool) {
+        TORCH_CHECK(mask_value == 0 || mask_value == 1, "Mask tensor can take 0 and 1 values only");
+      }
+      if (mask_value) {
+        int64_t offset = *(int64_t*)(mask_prefix_sum + strides[3] * i);
+        int64_t offset_bytes = (offset - 1) * sizeof(scalar_t);
```

Comment:
`std::partial_sum` functionally equals to `std::inclusive_scan` so we have to make adjustment here.
Once we are able to use `std::exclusive_scan`, this line needs to be:
`int64_t offset_bytes = offset * sizeof(scalar_t);`